### DISCUSSION TITLE
ParagonIE\CipherSweet\KeyProvider\StringProvider::__construct(): Argument #1 ($rawKey) must be of type string, null given

### DISCUSSION THREAD

**Discussion by Astelll** [spatie/laravel-ciphersweet#13] [Q&A]
Hello!

I'm trying to encrypt data in my database by command:
`php artisan ciphersweet:encrypt <your-model-class> <generated-key>`
`php artisan ciphersweet:encrypt "App\Models\User" da61321c8814fe47f6f647f26330bc34f88fd8c32c1df25af4f0b3a30dff00e9`

but I receive such error
`ParagonIE\CipherSweet\KeyProvider\StringProvider::__construct(): Argument #1 ($rawKey) must be of type string, null given, called in /var/www/fastuser/data/www/api.dev.clanhall.net/vendor/spatie/laravel-ciphersweet/src/LaravelCipherSweetServiceProvider.php on line 52`

![image](https://user-images.githubusercontent.com/11922651/181840469-0e4ef03c-cc91-4171-b62b-7984d2c6f251.png)

What am I doing wrong and how can I try to fix it?
All instructions from readme were executed

**Comment by Astelll** [comment]
I asked this question at https://github.com/paragonie/ciphersweet/ project in [this thread](https://github.com/paragonie/ciphersweet/issues/75), and got such answer:
![image](https://user-images.githubusercontent.com/11922651/189483430-0d3c4496-d408-4320-9a2a-12bc9a42e063.png)

**Comment by mgkimsal** [comment]
Adding on to this - it seems you need to set the key in .env first, even though the docs say not to.  It's looking for the value of the env key, and it's null until you set it.  The instructions likely need to be updated.

**Reply by mikerockett** [reply]
Bumped into this as well. Cannot run `php artisan ciphersweet:generate-key` until this is done, which is rather counter-intuitive. (And, it isn't obvious, but they key needs to be 32-characters long.)

**Comment by freekmurze** [comment]
@riasvdv could you have a look and update the readme if necessary?

**Comment by yormy** [comment]
I can confirm, just generating the key first (it does work), then copying that key to the env and then encrypt the model with the key like the docs mention works  like a charm

**Reply by AvenueShops** [reply]
I can also confirm that putting the key into the .env first before encrypting is required.

Additionally, if you're updating the readme, it might be useful for people to know that they may need to first create and run a migration on their encrypted model table fields to increase the size to accommodate the encryption value, since its likely larger than the original value.

**Reply by mgkimsal** [reply]
PR submitted

https://github.com/spatie/laravel-ciphersweet/pull/40

**Reply by mgkimsal** [reply]
PR merged with docs in correct order :)
